lib/puppet/ssl/state_machine.rb

```diff
@@ -84,6 +98,51 @@ def next_state
         to_error(_('Could not download CA certificate: %{message}') % { message: e.message }, e)
       end
     end
+
+    private
+
+    def needs_refresh?(now, last_update)
+      return true if last_update.nil?
+
+      ca_ttl = Puppet[:ca_refresh_interval]
+      return false unless ca_ttl
```

Comment:
I can't remember; what does `0` mean in this context?

Reply:
Setting it to `0` effectively means refresh every run, because of this comparison:

```ruby
now.to_i > last_update.to_i + ca_ttl
```

where I'm assuming `Time.now` is going to be strictly greater (in seconds) than the last update.

In order to disable the refresh, you'd have to set it to a very large integer value (`10y`). Another option is to define the settings as type `:ttl` instead of `:duration`, as `:ttl` accepts the value `unlimited`. For example, that what we use for `environment_timeout`.

Reply:
One quirk of settings is that you can programmatically set the value to `nil`:
```
irb(main):003:0> Puppet[:crl_refresh_interval] = nil
=> nil
```
But you can't do that in `puppet.conf`, as puppet always parses a setting with an unspecified value as an empty string:

```
$ grep crl_refresh_interval ~/.puppetlabs/etc/puppet/puppet.conf
crl_refresh_interval = 

$ bx puppet config print all
Error: Invalid duration format '""' for parameter: crl_refresh_interval
```

Reply:
Ok, seems like something we will want to specify in the docs, or in the `:desc` field. But that can happen asynchronously from this PR.

Reply:
Makes sense, I'll update the description for CA and CRL refresh interval